### ISSUE TITLE
[BUG][V3] Stuck on Unknown, waiting for webserver for very long same after Revo & clean install + not working anyway

### ISSUE THREAD

**Issue by Mozoloa** [bitfocus/companion#2405]
### Is this a bug in companion itself or a module?

- [X] I believe this to be a bug in companion

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When I launch companion, the little launcher window comes up and it stays on "Unknown, waiting for webserver" for a solid 3min, 
![image](https://user-images.githubusercontent.com/20817233/229281837-bc7f2955-db03-4788-a3c1-1717d62172c5.png)

Then it says Running, but when clicking "launch GUI", the page loads for a very long time (more than a minute)
![image](https://user-images.githubusercontent.com/20817233/229282193-20b4912b-18c8-4f86-a41f-77f1d2741a9f.png)

This is obviously a pain for module dev, but I've tried on a clean install with no "dev modules path" and it's the same.

This wasn't like that a few weeks ago

### Steps To Reproduce

1. Launch companion
2. See that's stuck on Unknown
3. Wait 3 minutes
4. Launch GUI
5. Wait 1 min
6. Test your module then make a change
7. go to 1.

### Expected Behavior

Having this process happen almost instantly like it used to be

### Environment (please complete the following information)

```markdown
- OS:Windows 11 Pro 22H2
- Browser: Brave v1.49.132 with Chromium 111.0.5563.147
- Companion Version: 3.0.0+5768-beta-8204b291
```

### Additional context

I had 2.4.2 running before, I installed 3 and the problems started. So I uninstalled, then reinstalled, no change. Uninstall with revo, delete everything in appdata, restart, reinstall, no chance.

**Comment by Mozoloa** [comment]
Let me add that I'm starting from scratch as well, no module loaded, since I deleted roaming/companion and all related reg keys

**Comment by Mozoloa** [comment]
Update: After another full revo uninstall + reg & appdata clean, it's still the same problem. May I add that quitting the app while it's in the unknown state takes age, or straight up doesn't work

**Comment by dnmeid** [comment]
possibly related: #2382
I've not seen this at least with macOS 10.15 and windows 10

**Comment by Mozoloa** [comment]
I've had this for a while, but it's really on and off, no obvious reason at all, and it's happening right now as I nuked everything, so clean install with 0 modules

**Comment by Julusian** [comment]
Are you using any antivirus software? If so, could you try pausing/disabling that to see if it helps?

I can't reproduce this on Windows 10, Ubuntu 22.04 or Windows 11 (21H2, the machine won't update to newer builds)

**Comment by Mozoloa** [comment]
Nope, just the windows default one I guess, but right now companion opens instantly, as I said it's seemingly random

**Comment by stevespaw** [comment]
@Julusian  "Windows 11 (21H2, the machine won't update to newer builds)"
Make a USB drive with [RUFUS](https://rufus.ie/en/) and do the update from it. It will work just fine :-) .

**Comment by Julusian** [comment]
@stevespaw Microsoft won't let me download an iso, so I can't use Rufus.
And using the windows media creation tool also fails because unsupported cpu.

I might be able to try with running windows in a vm though

**Comment by Mozoloa** [comment]
For a while this stopped happening but since I updated to 3 RC1 today on windows, it's happening again, exact same symptoms

**Comment by obcocav** [comment]
@Julusian .  Similar issue as also in #2382 is back.  Maybe a memory leak involved also?  The performance on 3.0 is not near as good as V2 for me but the last few builds it gets slower and slower over run time until it finally is unusable.

**Comment by obcocav** [comment]
Build 5975 is unusable after only a few minutes…trying out the hopeful fix for a vmix issue and unable access the GUI after a short while.

**Comment by SHCTV** [comment]
I'm also running into this similar issue with Companion v3.1.2 on my Macbook Pro 2023 w/ Apple M2 chip running macOS 13.5.2

I'm able to open sometimes on the admin profile but sometimes gets hung up that Unknown message.  If I switch the GUI interface sometimes that helps it reconnect but is inconsistent.

Also, I'm trying to run it from another profile on this laptop for users I don't want to have access to the Admin profile and the app won't even open on that regular profile.  I've tried uninstalling, restarting and installing the new stable version with no luck.

Thanks in advance if anyone has any insight.

**Comment by Mozoloa** [comment]
3.2.2 on a clients pc, it still does it, more than a minute to launch 

![image](https://github.com/bitfocus/companion/assets/20817233/4e7e8396-9a53-44b7-ab1f-78e05890343d)

**Comment by Campbellkid** [comment]
We are having the same issue when trying to update from Companion 2.4.2 to 3.3.1 on one of our windows 10 machines. I have never been able to get to go past the "Unknown Waiting for webserver...".  
![companion pic 1](https://github.com/user-attachments/assets/07d940e6-4979-4406-bf04-c82fa423a952)

**Comment by Julusian** [comment]
> We are having the same issue when trying to update from Companion 2.4.2 to 3.3.1 on one of our windows 10 machines. I have never been able to get to go past the "Unknown Waiting for webserver...". 

Could you share the log file with me? Hopefully that will give an indication of what the problem is.
You can click on the tray icon and use the open config folder option, the log files are somewhere around there

**Comment by ldickens** [comment]
I have this issue pop up using version 3.4.3. It has been some time since I updated and when completing this update today the same "Unknown: Waiting for webserver" error is showing. When I click on the "Open Config folder" option in the tray icon rmb menu nothing happens. 

Do you have a directory I can look in for some logging info?

**Comment by Julusian** [comment]
@ldickens 
if youre using windows the logs are in `%APPDATA/companion/logs`. On macos, they're in `Library/Application Support/companion/logs` inside your user folder.

It would be good to get a copy of the logs, this issue is pretty stuck as we don't have a way of reproducing and don't have any indication of what the might be causing this

**Comment by Campbellkid** [comment]
I found somewhere in the forums that the port changed from 2.4 to 3.x.  I changed the port number to 8484 and then it launched correctly.  That was frustrating.

Greg
________________________________

**Comment by ldickens** [comment]
@Julusian  It does look  like it could be a permissions issue. I am trying this on an office network, although I have been told that there should be no restricting permissions. Log below.[companion-2024-11-06.0.log](https://github.com/user-attachments/files/17653455/companion-2024-11-06.0.log)

**Comment by Julusian** [comment]
Have you tried changing the port it is listening on? Maybe something else is using port 8000, blocking companion from listening on it 
That is doable in that red launcher window

**Comment by ldickens** [comment]
I did try port 8484 as stated above. I can troubleshoot some more tomorrow and see if I can uncover anymore info.

**Comment by pilgrims1** [comment]
I had same issue in my computer that already sucessfully using Companion old versions.
But Now I found way to overcome.
Key point was Antivirus program.
My new latest COMPANION program (ver.3.5) runing well after uninstall anti virus program. ( in my case.... it was "Ahnlab safe transaction".)

**Comment by jcandrews310** [comment]
Hi all. I'm having this same issue in 4.3.1. Did the port change to 8484 with no change. Also tried 8888 and 8000 for fun, also with no change. 
I downgraded to the last version we had saved, 3.2.1, and it worked as expected. 
Any additional advise would be appreciated.
Thanks in advance.

**Comment by BambooStrategic** [comment]
[companion-2026-05-05.0.log](https://github.com/user-attachments/files/27422400/companion-2026-05-05.0.log)

Incredibly frustrated.  I was successfully using Companion 2 hours ago, restarted  my PC and then this issue started.  I tried to change the port, but the load is so fast that i can't get it to 8484 before the error.  I uninstalled, re-installed, tried earlier versions, reisntalled 4.3.1, and now nothing.  My daily production has relied upon companion for 2 years without issues.  This is a major issue at this point with less than 10 hours till production of a live morning show.

**Comment by BambooStrategic** [comment]
I'm at another office, loaded companion on my PC, and I'm getting the same issue- two PC's , two different locations.  I tried to log into  127.0.0.1:8000 directy and 8484, and no luck.  I am running Companion on a Mac mini  as my alternate.  I tried previous versions at this location too, and nothing works.

**Comment by Alan-pas** [comment]
I have the same issue.  On one pc, v4.3.2 works with no trouble.  On another pc, the launch window displays "Unknown," "waiting for webserver."  v2.4.2 works flawlessly on both pcs.  Both pcs are on the same wired network; both have similar firewall and antivirus settings.

**Comment by Julusian** [comment]
@BambooStrategic from that log file:

> 307C0200:error:80000005:system library:BIO_new_file:Input/output error:c:\ws\deps\openssl\openssl\crypto\bio\bss_file.c:67:calling fopen(C:/Program Files/Zoom Rooms Custom AV Controller, rb)

This seems to be a semi-common issue, as `Zoom Rooms Custom AV Controller` has incorrectly set an environment variable that is causing our instance of node to crash

**Comment by Julusian** [comment]
I think I have a fix for this recent crashing, but haven't been able to verify it.
It will be in 4.3.3 (maybe next week), or you can use this build https://s4.bitfocus.io/builds/companion/companion-win64-4.3-9227-ca-fix-4d64e720cd.exe which is 4.3.2 plus this fix.

**Comment by BambooStrategic** [comment]
Thanks,

I was able to manage this by removing the variable from the system.  But
hopefully the new build prevents this for others as well

Paul
